### PR TITLE
cli: add tests for running against absent server

### PR DESCRIPTION
**Pull request by tbg**
The caveat here is that those might be flaky if another process
opens the port during stressrace, but we will see.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9938)

### CODE REVIEW

**Comment by bdarnell** [comment]
LGTM. I think you could also test this by trying to connect to port 0, which is guaranteed to fail.

**Comment by tschottdorf** [comment]
Good idea. It changes the error messages though, so I added a comment for any future reader looking at this test if it's flaky.

File: pkg/cli/cli_test.go [hunk 1]
```diff
@@ -205,7 +205,41 @@ func TestQuit(t *testing.T) {
...
+	// NB: if this test is ever flaky due to port reuse, we could run against
+	// :0 (which however changes some of the errors we get).
+	// One way of getting that is:
+	//	c.TestServer.Cfg.AdvertiseAddr = "127.0.0.1:0"
+
+	for i, test := range []struct {
+		cmd, expOutPattern string
+	}{
+		{`debug kv inc a`,
+			`increment failed: roachpb.Batch RPC failed: rpc error: code = 14 desc = grpc: the connection is unavailable`},
```
Comment (tamird) [inline pkg/cli/cli_test.go]: don't you want the "grpc wrap" of this error?

Comment (tschottdorf) [inline pkg/cli/cli_test.go]: Good point. #9944.

**Pull request merged**